packages/web/app/views/administration/users/UserProfileModal.jsx

```diff
@@ -63,6 +63,22 @@ const validationSchema = yup.object().shape({
   email: foreignKey(<TranslatedText stringId="validation.required.inline" fallback="*Required" />)
     .email()
     .translatedLabel(<TranslatedText stringId="admin.users.email.label" fallback="Email" />),
+  newPassword: yup.string().nullable(),
+  confirmPassword: yup
+    .string()
+    .nullable()
+    .test(
+      'passwords-match',
+      <TranslatedText stringId="validation.passwords.mismatch" fallback="Passwords do not match" />,
+      function(value) {
+        const { newPassword } = this.parent;
+        // Only validate if both passwords are provided
+        if (!newPassword && !value) return true;
+        if (newPassword && !value) return false;
+        if (!newPassword && value) return false;
+        return newPassword === value;
```

Comment:
the last two ifs aren't really needed, they're handled equally well by this predicate